### PR TITLE
Allow embedding macOS bundle targets into macOS/Catalyst apps

### LINKED ISSUE #3712: Tuist dont support using macOS bundle in a mac Catalyst target
**Describe the bug**
I'm using tuist for our app, and our app enable mac catalyst recently. 
However when I try to set a macOS bundle as a dependency of my main target, tuist just echo a error:

```
Target TargetA has a dependency with target TargetB of type bundle for platform 'macOS' which is invalid or not supported yet.
Fatal linting issues found
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```
in which TargetA is our main app target(which is a catalyst target), TargetB is a macOS target statemented as below
```swift
    public static let TargetB = Target(
        name: "TargetB",
        platform: .macOS,
        product: .bundle,
        bundleId: bundleId,
        deploymentTarget: .macOS(targetVersion: "10.15"),
        infoPlist: .default,
        sources: [
        ]
}
```

**Expected behavior**
Be able to set TargetB as a dependency of TargetA.

**Desktop (please complete the following information):**

- OS: [macOS]
- Version [1.48.0]

### LINKED ISSUE #6798: macCatalyst app with macOS bundle (critical)
### What problem or need do you have?

How to create a target with `destinations: [.macCatalyst]` and dependency a macOS bundle. currently I am getting:

```
Target Example has platforms 'iOS' and product 'application' and depends on target MacBundle of type 'bundle' and platforms 'macOS' which is an invalid or not yet supported combination.
```

This is a very common situation: If you want to call some AppKit APIs, you create a macOS bundle and a protocol, then call them from iOS.

### Potential solution

_No response_

### macOS version

14.6.1

### Tuist version

4.28.1

### Xcode version

16.0.0

### LINKED ISSUE #6806: Target DummyApp has platforms 'macOS' and product 'application' and depends on target MacBundle of type 'bundle' and platforms 'macOS' which is an invalid or not yet supported combination.
### What happened?

Can't set as dependency a `bundle`.

### How do we reproduce it?

[TuistBug3.zip](https://github.com/user-attachments/files/17243035/TuistBug3.zip)

1. tuist generate

### Error log

Target DummyApp has platforms 'macOS' and product 'application' and depends on target MacBundle of type 'bundle' and platforms 'macOS' which is an invalid or not yet supported combination.

### macOS version

14.6.1

### Tuist version

4.28.1

### Xcode version

16.0.0

### PR DESCRIPTION
**Pull request by foyoodo**
Resolves #3712 
Resolves #6798 
Resolves #6806

Needs <https://github.com/tuist/XcodeGraph/pull/86>

### Short description 📝

Allow embedding macOS bundle targets into macOS apps and Mac Catalyst apps.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/get-started) for reference).

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by foyoodo** [comment]
Should we validate during the linting phase whether the iOS target includes Catalyst, or simply skip creating the embed plugin if Catalyst is not included?

File: Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift
Comment (foyoodo) [inline Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:755]: I’m not sure where it would be most appropriate to add the same logic for applying filters to "Target Dependencies", as changes to these two sections in Xcode are always synchronized. 

However, even without handling this, it won’t affect the product archived by Xcode.

Comment (foyoodo) [reply Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:755]: Alternatively, we could perform the validation during the linting phase.

File: Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift
Comment (fortmarek) [inline Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:736]: nit: it should _plugin_ and _plugIn_. I find the casing a bit weird. Would you mind changing that?

**Review by fortmarek — COMMENTED** [review]
Hey @foyoodo 👋 

Thanks a lot for the PR. Would you mind creating a fixture and an acceptance test? The fixture would help a ton in testing out this feature works properly.

File: Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift
Comment (foyoodo) [reply Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:736]: Good catch! I hadn’t noticed that. I’ll switch to *plugin* for consistency.

**Comment by foyoodo** [comment]
> Hey @foyoodo 👋
> 
> Thanks a lot for the PR. Would you mind creating a fixture and an acceptance test? The fixture would help a ton in testing out this feature works properly.
> 
> We'll also need to review the PR to `XcodeGraph` as I saw you point to your own fork.

Before adding the acceptance test, I need to build it. Please review the new [PR](https://github.com/tuist/tuist/pull/7134).

File: Sources/TuistCore/Graph/GraphTraverser.swift
Comment (fortmarek) [inline Sources/TuistCore/Graph/GraphTraverser.swift:1444]:
```suggestion
    private func canDependencyEmbedAsPlugin(dependency: GraphDependency) -> Bool {
```

**Review by fortmarek — COMMENTED** [review]
This looks good! I'd make the fixture a bit leaner if we could. Otherwise, just a couple of nits after which we can merge 🙂

File: Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift
Comment (foyoodo) [reply Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:743]: This literal is the default provided by Xcode, and modifying it doesn’t appear to have any negative impact. Should we consider making this change?

Comment (fortmarek) [reply Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift:743]: I see, if that's the default, we can keep it 👍

File: Package.swift
Comment (fortmarek) [inline Package.swift:512]: You can now update this to `1.1.0`: https://github.com/tuist/XcodeGraph/releases/tag/1.1.0

**Review by fortmarek — APPROVED** [review]
LGTM once this PR points to the `tuist/XcodeGraph` instead of the fork 🙂

File: Package.swift
Comment (foyoodo) [reply Package.swift]: Done! Thank you for taking the time to review this PR :)

**Review by fortmarek — APPROVED** [review]
:shipit:

**Pull request merged**
